src/thunderbots/software/ai/navigator/obstacle/obstacle.cpp

```diff
@@ -0,0 +1,117 @@
+    double tick_length)
+{
+    double radius_cushion =
+        getRadiusForHexagon(ROBOT_MAX_RADIUS_METERS + additional_radius_cushion_buffer);
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length + additional_velocity_cushion_buffer);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
+}
+
+
+Obstacle Obstacle::createRobotObstacleFromPositionAndRadiusAndVelocity(
+    Point position, double radius_cushion, Vector velocity_cushion_vector,
+    bool enable_velocity_cushion)
+{
+    if (enable_velocity_cushion)
+    {
+        Vector velocity_direction_norm_radius =
+            velocity_cushion_vector.norm(radius_cushion);
+        return Obstacle(Polygon(
+            {// left side of robot
+             position + velocity_direction_norm_radius.rotate(Angle::quarter()),
+             // back left of robot
+             position + velocity_direction_norm_radius.rotate(Angle::ofDegrees(150)),
+             // back right of robot
+             position + velocity_direction_norm_radius.rotate(Angle::ofDegrees(210)),
+             // right side of robot
+             position + velocity_direction_norm_radius.rotate(Angle::threeQuarter()),
+             // right side velocity cushions
+             position + velocity_direction_norm_radius.rotate(Angle::threeQuarter()) +
+                 velocity_cushion_vector,
+             // left side velocity cushions
+             position + velocity_direction_norm_radius.rotate(Angle::quarter()) +
+                 velocity_cushion_vector}));
+    }
+    else
+    {
+        // force the robot to face in +x direction
+        Vector facing_direction_norm_radius = Point(1, 0).norm(radius_cushion);
+        return Obstacle(Polygon(
+            {// left side of robot
+             position + facing_direction_norm_radius.rotate(Angle::quarter()),
+             // back left of robot
+             position + facing_direction_norm_radius.rotate(Angle::ofDegrees(150)),
+             // back right of robot
+             position + facing_direction_norm_radius.rotate(Angle::ofDegrees(210)),
+             // right side of robot
+             position + facing_direction_norm_radius.rotate(Angle::threeQuarter()),
+             // front right velocity cushions
+             position + facing_direction_norm_radius.rotate(Angle::ofDegrees(330)),
+             // front left velocity cushions
+             position + facing_direction_norm_radius.rotate(Angle::ofDegrees(30))}));
+    }
+}
+
+double Obstacle::getRadiusForHexagon(double radius)
+{
+    // return radius so that centre to side distance is at least radius
+    return radius * 4 / std::sqrt(3);
```

Comment:
I don't really understand the math here. What are `4` and `3`?

Reply:
added more comments